src/refiners/training_utils/config.py

```diff
@@ -22,6 +22,9 @@
 
 class TrainingConfig(BaseModel):
     device: str = "cpu"
+    automatic_mixed_precision: bool = (
+        True  # Enables automatic mixed precision which allows float32 gradients while working with lower precision.
+    )
     dtype: str = "float32"
```

Comment:
Since AMP is on by default (opt-out), and given that this `dtype` config is used jointly given the `autocast(dtype=self.dtype, enabled=self.config.training.automatic_mixed_precision)` call, any reason not to pick a sane default here? Namely: either `float16` or `bfloat16`. Might also deserve a comment.

(accelerate even offers to "Choose from ‘no’,‘fp16’,‘bf16 or ‘fp8’.")

Reply:
We could set it to False by default, and even though float32 + amp doesn't do much, I don't see this as a big issue; I would rather keep the API straightforward.